packages/desktop-client/src/components/reports/spreadsheets/custom-spreadsheet.ts

```diff
@@ -111,20 +115,26 @@ export function createCustomSpreadsheet({
       ).then(({ data }) => data),
     ]);
 
-    const months = monthUtils.rangeInclusive(startDate, endDate);
+    const rangeInc =
+      interval === 'Monthly' ? 'rangeInclusive' : 'yearRangeInclusive';
+    const format = interval === 'Monthly' ? 'monthFromDate' : 'yearFromDate';
+    const intervals = monthUtils[rangeInc](
+      monthUtils[format](startDate),
+      monthUtils[format](endDate),
+    );
 
     let totalAssets = 0;
     let totalDebts = 0;
 
-    const monthData = months.reduce((arr, month) => {
-      let perMonthAssets = 0;
-      let perMonthDebts = 0;
+    const intervalData = intervals.reduce((arr, inter) => {
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏
```suggestion
    const intervalData = intervals.reduce((arr, interval) => {
```